### PR TITLE
feat(ARCH-737): add ESM support (recovery PR)

### PR DESCRIPTION
**Pull request by smouillour**
**What is the problem this PR is trying to solve?**

our lib are in the CommonJS format.
We want to move forward by using vite as bundler which is the common mainstream tool today to work on react.
To achieve that we need to modernize our libraries  and start to push ESM format.

Original PR: https://github.com/Talend/ui/pull/5263

**What is the chosen solution to this problem?**

- [x] add script to start the playground with vite
- [x] add output esm configuration to all ui libraries (packages and fork)

**Please check if the PR fulfills these requirements**

- [ ] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Review by Gbacc — APPROVED** [review]
Hard to take a look at every single piece but it looks good

File: packages/components/package.json [hunk 1]
```diff
@@ -78,22 +89,23 @@
...
     "@testing-library/react-hooks": "^8.0.1",
+    "@testing-library/user-event": "^14.5.2",
     "@types/classnames": "^2.3.1",
     "@types/d3": "^7.4.3",
     "@types/jest": "^29.5.12",
     "@types/lodash": "^4.14.202",
     "@types/prop-types": "^15.7.11",
     "@types/react": "^18.2.60",
     "@types/react-dom": "^18.2.19",
+    "cpx2": "^7.0.1",
```
Comment (jmfrancois) [inline packages/components/package.json:108]: I am not sure to understand why you need cpx2 here ?

File: packages/design-docs/src/status.json [hunk 2]
```diff
@@ -0,0 +1,699 @@
+{
+  "accordion": {
```
Comment (jmfrancois) [inline packages/design-docs/src/status.json:2]: can you remove it from the design system ?

Comment (jmfrancois) [inline packages/design-docs/src/status.json:2]: This create an issue: it duplicates the status of the sync. it will makes it too hard to update.

**Review by jmfrancois — APPROVED** [review]

**Review by VolodymyrKovalM — APPROVED** [review]

**Pull request merged**